src/nvim/mouse.c

```diff
@@ -412,6 +412,18 @@ bool do_mouse(oparg_T *oap, int c, int dir, int count, bool fixindent)
 
   // Middle mouse button does a 'put' of the selected text
   if (which_button == MOUSE_MIDDLE) {
+    // handled outside any specific mode
+    if (tab_page_click_defs != NULL) {  // only when initialized
+      // Check for clicking in the tab page line.
+      if (mouse_grid <= 1 && mouse_row == 0 && firstwin->w_winrow > 0) {
+        if (is_click && cmdwin_type == 0 && mouse_col < Columns) {
+          in_tab_line = true;
+          c1 = tab_page_click_defs[mouse_col].tabnr;
+          handle_tabclose(c1);
+        }
+        return false;
+      }
+    }
     if (State == MODE_NORMAL) {
```

Comment:
use a blank line to separate this "group" of code, just like the other cases in this same function